src/content/profile-data.js

```diff
@@ -133,11 +133,21 @@ export function defaultThreadOrder(threads: Thread[]) {
 }
 
 export function filterThreadByImplementation(thread: Thread, implementation: string): Thread {
-  const { funcTable } = thread;
+  const { funcTable, stringTable } = thread;
 
   switch (implementation) {
     case 'cpp':
-      return _filterThreadByFunc(thread, funcIndex => !funcTable.isJS[funcIndex]);
+      return _filterThreadByFunc(thread, funcIndex => {
+        // Return quickly if this is a JS frame.
+        if (funcTable.isJS[funcIndex]) {
+          return false;
+        }
+        // Try to filter out jitcode frames, they are named like 0xNNNNNNN where N is some
+        // number, and they don't have any resource associated with them.
```

Comment:
Maybe add: "Regular C++ functions are associated with a resource that describes the shared library that these C++ functions were loaded from. Jitcode is not loaded from shared libraries but instead generated at runtime, so Jitcode frames are not associated with a shared library and thus have no resource."